plaso/parsers/fseventsd.py

```diff
@@ -159,8 +153,23 @@ def ParseFileObject(self, parser_mediator, file_object, **kwargs):
     Raises:
       UnableToParseFile: when the header cannot be parsed.
     """
-    page_header = self._ParseDLSPageHeader(file_object)
+    page_header_map = self._GetDataTypeMap('dls_page_header')
```

Comment:
Why not collapse the first page header check into the loop?

Reply:
There's several things that need to be set up only once the first header has been read. I think this way is easier to read.